fatrace.c

```diff
@@ -599,6 +710,14 @@ main (int argc, char** argv)
     struct sigaction sa;
     struct timeval event_time;
 
+    /* use a UTF-8 locale if possible so is_valid_string works as expected */
+    locale_t utf8 = newlocale(LC_CTYPE_MASK, "C.UTF-8", NULL);
+    if (!utf8) utf8 = newlocale(LC_CTYPE_MASK, "en_US.UTF-8", NULL);
+    if (utf8)
+        uselocale(utf8);
+    else
+        warnx ("Could not switch to a UTF-8 locale");
```

Comment:
Hmm.. Either mbrtoc32 requires setting an UTF-8 locale, then the failure path can't just be a warning -- it would need to modify the behaviour of `is_valid_utf8()`. Or mbrtoc32 doesn't care about the locale (which sounds plausible, as UTF-8 can be decoded independently of the locale -- it doesn't get written to screen, etc.), then this code is unnecessary.

Reply:
IIUC, mbrtoc32 does not require setting a UTF-8 locale, but it does care. If we fail to set the locale, mbrtoc32 will decode the current locale. This does mean that we risk invalid output. For example, if current locale is latin-1 and neither UTF-8 locale is installed, we'll get this warning and `is_valid_mb_string` will return true on valid latin-1 strings even if they are invalid UTF-8. `print_json_str` will then produce invalid JSON. What do you think, should we exit with error if we're unable to use UTF-8?

Reply:
>  If we fail to set the locale, mbrtoc32 will decode the current locale. 

The fallback could be to just set the "C" locale, which would/should then mean it only accepts ASCII. That should be alright, and compatible with JSON.

We don't have to error out in this case, you'd just get the *_raw byte arrays then.